build.ps1

```diff
@@ -254,17 +254,21 @@ function Build-WiXProject()
 {
   [CmdletBinding(PositionalBinding = $false)]
   param(
-    [Parameter(Position = 0)]
+    [Parameter(Position = 0, Mandatory = $true)]
     [string]$FileName,
+    [Parameter(Mandatory = $true)]
+    [hashtable]$Arch,
     [hashtable]$Properties = @{}
   )
 
   $Name = $FileName.Split('.')[0]
+  $ArchName = $Arch.VSName
 
   $Properties = $Properties.Clone()
+  TryAdd-KeyValue $Properties ProductArchitecture $ArchName
```

Comment:
This is worrisome.  The SDKs are architecture agnostic (they are basically headers and import libraries).

Reply:
But we have `sdk-x86.wxs`, `sdk-amd64.wxs` and `sdk-arm64.wxs` which are selected in `sdk.wixproj` using:

```xml
  <ItemGroup>
    <Compile Include="sdk-$(ProductArchitecture).wxs" />
  </ItemGroup>

Reply:
Ah, that key is different from the actual key itself of what the architecture is that the MSI supports!

Reply:
I didn't understand? It seems like `ProductArchitecture` is the only knob we have to choosing which msi we're building:

```
    <ProductArchitecture Condition=" '$(ProductArchitecture)' == '' ">amd64</ProductArchitecture>
    <ProductArchitecture>$(ProductArchitecture)</ProductArchitecture>
```

```
  <PropertyGroup Condition=" '$(ProductArchitecture)' == 'amd64' ">
    <InstallerPlatform>x64</InstallerPlatform>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(ProductArchitecture)' == 'arm64' ">
    <InstallerPlatform>arm64</InstallerPlatform>
    <InstallerVersion>500</InstallerVersion>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(ProductArchitecture)' == 'arm' ">
    <InstallerPlatform>arm</InstallerPlatform>
    <InstallerVersion>500</InstallerVersion>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(ProductArchitecture)' == 'x86' ">
    <InstallerPlatform>x86</InstallerPlatform>
  </PropertyGroup>

Reply:
The `ProductArchitecture` indicates the architecture, but we don't have a `<Condition>` that prevents the installation on another architecture.